### PR TITLE
Saagar/find by

### PR DESCRIPTION
**Pull request by Saagar3690**
### **Changes**
1. Changed all endpoints in all routes, as needed, from findByIdAndX to findById, check, then X.
2. Modified to use async/await and try catch block
3. Documented user
4. Expanded consistency everywhere (i.e use eventId everywhere in routes/events.js instead of eventId in some places and id in others)
5. Modified some endpoints on Postman and added missing endpoints for user to Postman (makeOrganizer, removeOrganizer)
Screen Shot 2020-12-09 at 10 10 03 PM

### **Verify Tests Work After Changes**
Screen Shot 2020-12-09 at 10 05 29 PM

### **Verify POSTMAN Endpoints Work After Changes**
The two 404 response codes for users/makeOrganizer were because I sent the mongoose id instead of the firebase id and the 400 response code for users/add was because I forgot to send in a firebase id in request
Screen Shot 2020-12-09 at 10 41 36 PM

Screen Shot 2020-12-09 at 10 40 58 PM
Screen Shot 2020-12-09 at 10 41 11 PM

### CODE REVIEW

File: routes/bots.js [hunk 1]
```diff
@@ -132,18 +136,25 @@ botsRouter.route('/').post((req, res) => {
...
  */
-botsRouter.route('/addItem').post((req, res) => {
+botsRouter.route('/addItem').post(async (req, res) => {
 	const botId = req.body.botId;
 	const itemId = req.body.itemId;
 	const quantity = req.body.quantity;
 
 	if (!botId || !itemId || !quantity) {
-		res
+		return res
```
Comment (dvorjackz) [inline routes/bots.js:145]: Thank you for changing these, did you change all of them?

File: routes/events.js [hunk 2]
```diff
@@ -130,89 +147,122 @@ eventsRouter.route('/add').post((req, res) => {
...
 /**
  * Adds an admin id to the list of admin ids to an event specified by id
  */
-eventsRouter.route('/admins').put((req, res) => {
-	const { id, admin_id } = req.body;
+eventsRouter.route('/admins').put(async (req, res) => {
+	const { eventId, admin_id } = req.body;
 
-	if (!id) {
+	if (!eventId) {
```
Comment (dvorjackz) [inline routes/events.js:211]: Thanks for renaming all of these id's as well

File: routes/users.js [hunk 3]
```diff
@@ -4,15 +4,34 @@ const router = express.Router();
 
 let { User } = require('../models/user.model');
 
+/**
```
Comment (dvorjackz) [inline routes/users.js:7]: Nice

File: routes/users.js [hunk 4]
```diff
@@ -25,34 +44,96 @@ router.route('/add').post((req, res) => {
...
+
+	try {
+		let user = await User.findOne({ firebase_id: firebase_id });
+
+		if (!user)
+			return res
+				.status(404)
+				.json('Could not find user specified by firebase_id.');
+
+		user.isOrganizer = true;
```
Comment (dvorjackz) [inline routes/users.js:70]: No need to have two, let's just do one endpoint and pass in true/false so we don't have too many endpoints

File: routes/users.js [hunk 5]
```diff
@@ -25,34 +44,96 @@ router.route('/add').post((req, res) => {
...
+				.status(404)
+				.json('Could not find user specified by firebase_id.');
+
+		user.isOrganizer = false;
+		user.save();
+		res.json('User ' + req.body.firebase_id + ' was made not an organizer!');
+	} catch (err) {
+		console.log('Error: ' + err);
+		res.status(400).json(err);
+	}
```
Comment (dvorjackz) [inline routes/users.js:104]: Delete

**Review by dvorjackz — CHANGES_REQUESTED** [review]

**Comment by dvorjackz** [comment]
Great housekeeping, +1

File: routes/bots.js (hunk 1 above)
Comment (Saagar3690) [reply routes/bots.js:145]: Yes, I believe I changed all of them to include return.

**Review by Saagar3690 — COMMENTED** [review]
### **Changes**
Combined addOrganizer and removeOrganizer into just users/organizer and it takes in firebase_id and a boolean call isOrganizer.
Modified paths to have consistent pathId instead of id to follow other routes i.e botId, itemId, etc.
Screen Shot 2020-12-11 at 8 36 51 PM
Screen Shot 2020-12-11 at 8 36 58 PM

**Review by dvorjackz — APPROVED** [review]
Approved - fix the merge conflict and merge it in!

**Pull request merged**